packages/components/forms/src/FormControl/FormControlContext.tsx

```diff
@@ -0,0 +1,52 @@
+import { createContext, useContext } from 'react';
+
+export interface FormControlProps {
+  id?: string;
+  /**
+   * If `true` set the form control to the invalid state.
+   */
+  isInvalid?: boolean;
+  /**
+   * If `true` set the form control to be required.
+   */
+  isRequired?: boolean;
+  /**
+   * If `true` set the form control to the disabled state.
+   */
+  isDisabled?: boolean;
+
+  isReadOnly?: boolean;
+}
```

Comment:
Should this type live here? In the Context file?

IMO it would be better in a `types` file if we want to share it between both files